### PR TITLE
Made document changes according to Sphinx theme

### PR DESCRIPTION
**Pull request by vkathir82**
Hi Andy & Adilet,

This is the new PR without product build.

I have made changes as per the procedure followed by zos team by using the tools that can extract documentation from our Ansible modules into ReStructuredText, generate HTML using Sphinx and did the format with a Sphinx plugin. Please review it as this changes directory structure from what we have.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: e4a7fc163fd424309f8925caaa2858bfc28fc5ce
  * Invalid signoff. Commit message must end with  
    **Signed-off-by: .* <.*>**

Guidelines are available at https://github.com/zhmcclient/zhmc-ansible-modules/blob/master/CONTRIBUTING.rst

---
This message was auto-generated by https://gitcop.com

**Comment by andy-maier** [comment]
The `make docs` fails with:
```
$ if [[ $TRAVIS_PYTHON_VERSION == 3.8 && $PACKAGE_LEVEL == latest ]]; then make docs; fi
make: *** No rule to make target 'docs/conf.py', needed by 'build/docs/html/index.html'.  Stop.
The command "if [[ $TRAVIS_PYTHON_VERSION == 3.8 && $PACKAGE_LEVEL == latest ]]; then make docs; fi" exited with 2.
```
The file `docs/conf.py` still exists at that location, so I'm not sure I understand why this fails.

**Comment by andy-maier** [comment]
The commit message fails the gitcop check. Please fix.

**Review by andy-maier — CHANGES_REQUESTED** [review]
See my two comments

File: docs/templates/module.rst.j2 [hunk 1]
```diff
@@ -0,0 +1,1684 @@
+
```
Comment (andy-maier) [inline docs/templates/module.rst.j2:1]: This file looks completele different compared to what you have shown in the meeting today. PLease double check whether this is really the file you wanted to have here.

File: docs/make.bat [hunk 2]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (andy-maier) [inline docs/make.bat:1]: I suggest to remove this batch file. We should be working through the Makefile.

**Comment by andy-maier** [comment]
The ansible-doc-extractor package should be added to the dev-requirements.txt file.

**Comment by andy-maier** [comment]
The build of the docs using the Makefile in the docs directory has many warnings:
```
$ cd docs
$ make html
Running Sphinx v3.2.1
making output directory... done
WARNING: html_static_path entry '_static' does not exist
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 21 source files that are out of date
updating environment: [new config] 21 added, 0 changed, 0 removed
reading sources... [100%] modules/zhmc_virtual_function                                                                                          
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_adapter.rst:188: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_adapter.rst:191: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_adapter.rst:192: WARNING: Unexpected section title.

},
...
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_adapter.rst:193: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_adapter.rst:194: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_adapter.rst:196: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:126: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:129: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:130: WARNING: Unexpected section title.

},
...
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:131: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:133: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:136: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:137: WARNING: Unexpected section title.

},
...
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:138: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:139: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst:141: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:167: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:168: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:170: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:173: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:176: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:177: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:179: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:185: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst:187: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_hba.rst:145: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_hba.rst:147: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_nic.rst:144: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_nic.rst:146: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_partition.rst:214: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_partition.rst:216: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_storage_group.rst:159: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_storage_group.rst:161: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_storage_group_attachment.rst:138: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_storage_volume.rst:154: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_storage_volume.rst:156: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_user.rst:150: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_user.rst:152: WARNING: Unexpected indentation.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_virtual_function.rst:141: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_virtual_function.rst:143: WARNING: Unexpected indentation.
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_adapter.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_cpc.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_crypto_attachment.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_hba.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_nic.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_partition.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_storage_group.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_storage_group_attachment.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_storage_volume.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_user.rst: WARNING: document isn't included in any toctree
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/zhmc_virtual_function.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] modules/zhmc_virtual_function                                                                                           
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/modules_support.rst:19: WARNING: undefined label: core_supported (if the link has no caption the label must precede a section header)
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/modules_support.rst:25: WARNING: undefined label: network_supported (if the link has no caption the label must precede a section header)
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/modules_support.rst:59: WARNING: undefined label: modules_by_category (if the link has no caption the label must precede a section header)
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/modules_support.rst:61: WARNING: undefined label: intro_adhoc (if the link has no caption the label must precede a section header)
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/modules_support.rst:63: WARNING: undefined label: working_with_playbooks (if the link has no caption the label must precede a section header)
/Users/maiera/Projects/zhmcclient/repos/zhmc-ansible-modules/docs/source/modules/modules_support.rst:65: WARNING: undefined label: developing_modules (if the link has no caption the label must precede a section header)
generating indices...  genindexdone
writing additional pages...  searchdone
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 58 warnings.

The HTML pages are in build/html.
```

**Comment by andy-maier** [comment]
It is not clear how the rst files of the zhmc modules can be re-generated (and I always have a bad feeling when generated files are checked in to the repo). How is that done? Deleting the files in docs/sources/modules does not re-generate them.

File: docs/templates/module.rst.j2 (hunk 1 above)
Comment (vkathir82) [reply docs/templates/module.rst.j2:1]: @andy-maier I will remove this file template support for now as it still needs to be adjusted for our documentation.

File: docs/make.bat (hunk 2 above)
Comment (vkathir82) [reply docs/make.bat:1]: Sure, I will remove it

**Comment by andy-maier** [comment]
This PR has been replaced with PR #163

**Pull request closed without merging**